cmd/snap-confine/mount-support.c

```diff
@@ -304,6 +307,17 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 			// from that of its own snap.
 			sc_do_mount("none", dst, NULL, MS_REC | MS_SLAVE, NULL);
 		}
+		if (mnt->alias == NULL) {
+			continue;
+		}
+		// An alias of mnt->path is provided at another location.
+		// It should behave exactly the same as the original.
+		sc_must_snprintf(dst, sizeof dst, "%s/%s", scratch_dir,
+				 mnt->alias);
+		sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL);
+		if (!mnt->is_bidirectional) {
+			sc_do_mount("none", dst, NULL, MS_REC | MS_SLAVE, NULL);
+		}
```

Comment:
Reading this, when not MERGED_USR, nothing changes and we mount to mnt->path in scratch_dir/path, but if MERGED_USR, we do that and also mount mnt->path on scratch_dir/alias. The code assumes that the alias (ie, /media) exists and is not a symlink. It seems if /media is a symlink to /run/media this isn't needed and if /media doesn't exist at all, we need to create it. Am I missing something?

Reply:
It's almost correct, /media is never a symlink or is never absent and we cannot create it. /media is simply in the core snap (core16) and we will likely keep having it in core18. If it goes away we would need to alter this code but we'd know that well in advance as hard-coded /media mount would fail.

Reply:
Thanks for the clarification. So this is true for "/media" (and therefore correct as is), but it wouldn't be true for arbitrary values for alias/altpath. You mentioned 'if it goes away', and indeed, sc_do_mount() will die() if it isn't there. If a future core sets up a compatibility symlink, or snap-confine adds something new in this area such that alias -> path, then we'd double mount on the directory. What do you think about checking if this is a symlink?

Reply:
I can check and bail if this is a symlink, sure